### ISSUE TITLE
Using `code . --disable-extension xxx` in VSCode does not work when disabling extensions.

### ISSUE THREAD

**Issue by muzaisimao** [microsoft/vscode#170832]
When I use `code . --disable-extension dbaeumer.vscode-eslint` command in VSCode to disable the extension of the current folder/workspace it doesn't work.

Other attempts: also invalid when adding `--reuse-window` options.

Expected Result: Disable the specified extension.

![Example](https://user-images.githubusercontent.com/31382094/211271232-43f5d453-7591-4622-8036-f4abddfb93c2.gif)

**Comment by Cyclodex** [comment]
To extend it a little bit: These commands **do work**, when the workspace/folder is not yet open! (with disabled extensions)
Even we did not found any documentation about it...

**Comment by sandy081** [comment]
Right, this works only when the window is not opened. I do not think we send the CLI args to the opened window. Also this option is to disable the extension for the current session only. This will be reset when user restarts VS Code without this option. I will update the help doc.

**Comment by muzaisimao** [comment]
@sandy081  Thank you for your reply. I understand that the disabled will be reset when VS Code is restarted, but I still hope that the command will take effect in the current window, such as the following. Is it possible to solve it, or other alternatives?

```json
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "Open Refined Workspace",
            "type": "shell",
            "command": "code -r . --disable-extension dbaeumer.vscode-eslint",
            "problemMatcher": []
        }
    ]
}

```

**Comment by sandy081** [comment]
I am not sure if it is possible to update the args for the current window. 

CC @bpasero

**Comment by bpasero** [comment]
Only when you trigger a reload:

https://github.com/microsoft/vscode/blob/ff01a22a01a7aed8ae39a5ade70efc345769ef62/src/vs/platform/windows/electron-main/windowImpl.ts#L1054

**Comment by muzaisimao** [comment]
@bpasero Thank you. So for the user, there's no way to solve this problem, right?

**Comment by bpasero** [comment]
You cannot change the args of the current window as a user.

**Comment by Cyclodex** [comment]
> Right, this works only when the window is not opened. I do not think we send the CLI args to the opened window. Also this option is to disable the extension for the current session only. This will be reset when user restarts VS Code without this option. I will update the help doc.

I would like to add, that this feature `disable-extension` is not documented online somewhere right? I could not find anything about it... I just stumbled over it.

> You cannot change the args of the current window as a user.

Can an extension do this? Well it would reload the screen anyway, which is what we would like to prevent...

**Comment by bpasero** [comment]
Yeah not even internally you can change `args` without a window reload at the moment unless a specific component supports this of course, which depends on the use case.

**Comment by Cyclodex** [comment]
Question: The user can simply disable a extension per Click, even defined if globally or per workspace, is there no such API an extension could call doing this?

**Comment by bpasero** [comment]
@sandy081 would know

**Comment by sandy081** [comment]
> The user can simply disable a extension per Click, even defined if globally or per workspace, is there no such API an extension could call doing this?

There is no API for this - May I know why an extension has to do this? Related, Have you tried the new profiles feature that is in preview? - This can be enabled using setting `workbench.experimental.settingsProfiles.enabled` where you can create different profiles (settings, extensions, .... ) for different scenarios/use cases and switch seamlessly.

**Comment by muzaisimao** [comment]
Hi, @sandy081 
Yes, this is required in Vue, You can view the details here: https://vuejs.org/guide/typescript/overview.html#volar-takeover-mode.
I tried to use the profiles feature, but it seems that it cannot configure “Built-in Extensions”, and it is not very convenient to configure it once for each project, especially in teamwork.

**Comment by sandy081** [comment]
> I tried to use the profiles feature, but it seems that it cannot configure “Built-in Extensions”, 

You can configure enablement state of built in extensions in profiles. May I know if you tried this and if does not work and file an issue, because it is supposed to work.

> it is not very convenient to configure it once for each project, especially in teamwork.

You do not need to configure for each project. You configure a profile once and open the workspace using this profile.

**Comment by muzaisimao** [comment]
@sandy081 
I just tried it, the profiles feature can't meet this requirement, first of all, the exported configuration file is compressed, it's hard to read and modify., After formatting it, I found that it can't disable the extension.
 
When I switch between different profiles, it's more like uninstalling those extensions that are not selected, but built in extensions cannot be uninstalled.

I tried using a new profile, then disabled a built in extensions in VS Code and exported it, and when I imported it, the extension wasn't disabled.

**Comment by sandy081** [comment]
> I just tried it, the profiles feature can't meet this requirement, first of all, the exported configuration file is compressed, it's hard to read and modify., After formatting it, I found that it can't disable the extension.

This is not hand written and machine generated. 

> When I switch between different profiles, it's more like uninstalling those extensions that are not selected, but built in extensions cannot be uninstalled.

Yes, but built in extensions can be disabled and profile carry this information.

> I tried using a new profile, then disabled a built in extensions in VS Code and exported it, and when I imported it, the extension wasn't disabled.

Can you please share the profile?

**Comment by muzaisimao** [comment]
@sandy081 
Ok, here is the exported config file after disabling built-in extensions.

Extension id: vscode.typescript-language-features

VS Code: 1.75.0-insider

[Vue3.code-profile.zip](https://github.com/microsoft/vscode/files/10447529/Vue3.code-profile.zip)

**Comment by sandy081** [comment]
Thanks for the profile. I could repro the bug. Filed a separate issue tracking the bug so that it can be verified too.

https://github.com/microsoft/vscode/issues/171649
